src/extension.ts

```diff
@@ -9,22 +9,29 @@ import { CompletionProvider } from './providers';
 export let errorDiagnosticCollection = vscode.languages.createDiagnosticCollection("terraform-error");
 export let outputChannel = vscode.window.createOutputChannel("Terraform");
 
+const documentSelector: vscode.DocumentSelector = [
+    { language: "terraform", scheme: "file" },
+    { language: "terraform", scheme: "untitled" }
+];
+
 export function activate(ctx: vscode.ExtensionContext) {
     ctx.subscriptions.push(errorDiagnosticCollection);
 
     let formattingProvider = new FormattingEditProvider;
 
     // we need to manually handle save events otherwise format on autosave does not work
     ctx.subscriptions.push(vscode.workspace.onDidSaveTextDocument((doc) => formattingProvider.onSave(doc)));
-    vscode.languages.registerDocumentFormattingEditProvider('terraform', formattingProvider);
+    vscode.languages.registerDocumentFormattingEditProvider(documentSelector, formattingProvider);
 
     ctx.subscriptions.push(vscode.workspace.onDidChangeTextDocument(liveIndex));
 
     ctx.subscriptions.push(vscode.commands.registerCommand('terraform.validate', () => { validateCommand(); }));
     ctx.subscriptions.push(vscode.commands.registerCommand('terraform.lint', () => { lintCommand(); }));
 
-    ctx.subscriptions.push(vscode.languages.registerCompletionItemProvider("terraform", new CompletionProvider, '.', '"'));
+    ctx.subscriptions.push(vscode.languages.registerCompletionItemProvider(documentSelector, new CompletionProvider, '.', '"'));
 
     // index operations
-    initializeIndex(ctx);
+    if (vscode.workspace.rootPath) {
```

Comment:
just to double-check: this means that for the live-share guest `initializeIndex()` is not called? is there a way to get those features to work for a guest?

Reply:
The Live Share guest would automatically get all of the features that `initializeIndex()` is registering (e.g. go to definition, rename, document symbols). We could have updated the calls to `vscode.languages.register*` to take the same `DocumentSelector` I'm using here, but `initializeIndex` also does other things that would be driven by the host (e.g. listening for config changes, scanning for Terraform documents), and so it seemed simpler to just prevent calling it entirely for guests.